### PR TITLE
[DOCS-1582] updated titles, version numbers, and minor formatting issues for 2.3.x

### PR DESCRIPTION
**Pull request by HCloward**
### Review
@reviewer - TBD
@rainest @kallecaco3 suggested the Kong for kubernetes content be added to this doc for users upgrading to the next version of Kong. I selected some of the text from the [README.md](https://github.com/Kong/charts/blob/main/charts/kong/README.md#separate-admin-and-proxy-nodes) and [Upgrade.md](https://github.com/Kong/charts/blob/main/charts/kong/UPGRADE.md) docs that I think is relevant. Can you take a look and tell me if what I selected is appropriate or if more/less needs to be in this guide to help users upgrading on kubernetes? Thanks!

### Summary
The upgrade guides for upgrading from 2.2.x to 2.3.x and 2.3.x OSS to 2.3.x Enterprise need updating after the 2.3.x release.

### Reason
[DOCS-1582](https://konghq.atlassian.net/browse/DOCS-1582?focusedCommentId=38123&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-38123)

### Testing
[Link to KE upgrade guide updates](https://deploy-preview-2652--kongdocs.netlify.app/enterprise/2.3.x/deployment/upgrades/migrations/)

### CODE REVIEW

File: app/enterprise/2.3.x/deployment/upgrades/migrations.md [hunk 1]
```diff
@@ -15,65 +15,118 @@ to {{site.ee_product_name}}. See
...
   need to be installed on both the Control Plane and the Data Planes in Hybrid mode. Install the
   plugins on the Control Plane first, and then the Data Planes.
-  * The [Rate Limiting Advanced](/hub/kong-inc/rate-limiting-advanced) plugin does not
+* The [Rate Limiting Advanced](/hub/kong-inc/rate-limiting-advanced) plugin does not
     support the `cluster` strategy in hybrid mode. The `redis` strategy must be used instead.
-* If you are adding a new plugin to your installation, you need to run
-  `kong migrations up` with the plugin name specified. For example,
-  `KONG_PLUGINS=oauth2`.
 
+### Kong for kubernetes considerations
```
Comment (rainest) [inline app/enterprise/2.3.x/deployment/upgrades/migrations.md:80]:
```suggestion
### Kong for Kubernetes considerations
```

File: app/enterprise/2.3.x/deployment/upgrades/migrations.md [hunk 2]
```diff
@@ -15,65 +15,118 @@ to {{site.ee_product_name}}. See
...
+Kong version back, check the upgrade path recommendations for Kong open source or Kong Gateway.
+
+Although not required, users should upgrade their chart version and Kong version indepedently. 
+In the event of any issues, this will help clarify whether the issue stems from changes in 
+Kubernetes resources or changes in Kong.
+
+Users may encounter an error when upgrading which displays a large block of text ending with 
+`field is immutable`. This is typically due to a bug with the `init-migrations` job, which was 
+not removed automatically prior to 1.5.0. If you encounter this error, deleting any existing 
+`init-migrations` jobs will clear it.
```
Comment (rainest) [inline app/enterprise/2.3.x/deployment/upgrades/migrations.md:99]: Should be okay to clear this out. 1.5 is ancient and new installs should not use it.
```suggestion
```

File: app/enterprise/2.3.x/deployment/upgrades/migrations.md [hunk 3]
```diff
@@ -15,65 +15,118 @@ to {{site.ee_product_name}}. See
...
 
-### Migrating Databases for a Major or Minor Version Release {#migrate-db}
+For specific Kong for kubernetes version upgrade considerations, see
```
Comment (rainest) [inline app/enterprise/2.3.x/deployment/upgrades/migrations.md:101]:
```suggestion
For specific Kong for Kubernetes version upgrade considerations, see 
```

**Review by rainest — CHANGES_REQUESTED** [review]

**Review by rainest — APPROVED** [review]

**Review by lena-larionova — APPROVED** [review]

**Comment by lena-larionova** [comment]
@HCloward merging this in, as it was requested that this go live before the long weekend.

**Pull request merged**